lib/webrtc/call.js

```diff
@@ -145,6 +151,64 @@ MatrixCall.prototype.placeScreenSharingCall =
 };
 
 /**
+ * Play the given HTMLMediaElement, serialising the operation into a chain
+ * of promises to avoid racing access to the element
+ * @param {Element} HTMLMediaElement to play
+ * @param {string} Arbitrary ID to track the chain of promises to be used
+ */
+MatrixCall.prototype.playElement = function(element, queueId) {
+    if (MatrixCall.mediaPromises[queueId]) {
+        MatrixCall.mediaPromises[queueId] =
+            MatrixCall.mediaPromises[queueId].then(function() {
+                return element.play();
```

Comment:
I'm worried this is creating a memory leak.

We never nuke the promises in the map, and I don't know enough about how Q (in this case) stores chained promises to know whether or not they get cleaned up after use (I suspect not). This means we may be holding onto references to `element` in the middle of a promise chain.

I don't expect you to do anything right now for this (you can heap dump a few times if you really want to find out if it is leaking, the deltas on the `HTMLMediaElement` types will clearly show it), but heads up either way.